docs/docs/guides/js_apps/test.md

```diff
@@ -0,0 +1,175 @@
+---
+title: Testing Aztec.nr contracts with TypeScript
+---
+
+In this guide we will cover how to interact with your Aztec.nr smart contracts in a testing environment to write automated tests for your apps.
+
+To write your first smart contract, check out the [Aztec.nr getting started guide](../../getting_started.md).
+
+## Compile your contract and generate TS interface
```

Comment:
I think the compilation is covered in the other guide, so I would just link to that one. I would start with a prerequisites section like:

## Prerequisites

For this guide, we assume you:
- Have followed the getting started guide -> link to getting started
- Have a compiled contract -> link to compilation
etc

Basically leaving just the barebones "testing" section. This makes it easy to maintain as the instructions for compilation are in one place only